.github/workflows/ci.yml

```diff
@@ -32,4 +34,3 @@ jobs:
         uses: codecov/codecov-action@v1
         with:
           file: ./coverage.xml
-          fail_ci_if_error: true
```

Comment:
Removed this check for now as it is hitting a rate limit on Codecov and causing a [CI build failure](https://github.com/mitodl/edx-api-client/actions/runs/9127231186/job/25097088011). To bypass this limit, we need to use a Codecov repository upload token.

This issue also exists in other repositories. You can see the build step Upload coverage to CodeCov failing on the master branches, e.g., [mitxpro](https://github.com/mitodl/mitxpro/actions/runs/9080193803/job/24951049071) and [mitxonline](https://github.com/mitodl/mitxonline/actions/runs/9112869998/job/25053028809).

Could someone from the DevOps team take a look into [this](https://github.com/mitodl/edx-api-client/actions/runs/9127231186/job/25097088011)?

@blarghmatey ^^

Reply:
@mudassir-hafeez Upgrading `codecov/codecov-action@v1` to `codecov/codecov-action@v3` might fix this. Could you check that?